app/models/credit_card.rb

```diff
@@ -3,10 +3,10 @@ class CreditCard < ApplicationRecord
   attr_accessor :number
 
   before_create :set_last_four
-  validates :first_name, :last_name, :token, :month, :year, presence: true
+  validates :first_name, :last_name, :month, :year, presence: true
   belongs_to :user
 
   def set_last_four
-    self.last_four = number[-4..-1]
+    self.last_four = number[-4..-1].to_i
```

Comment:
this should be stored as text not an integer. What if my last four is 0077? then it would be saved as 77.